### PR TITLE
Adding example for creating a partitioned table

### LINKED ISSUE #5537: CREATE TABLE docs should have an example of creating a partitioned table
if i google for 'create partitioned table in presto' it is not very helpful, looking at specific https://prestosql.io/docs/current/sql/create-table.html does not help either. Need to query SELECT * FROM system.metadata.table_properties then find partitioned_by then look for example of how to specify it when creating a table

### PR DESCRIPTION
**Pull request by agrawalreetika**
Adding an example for creating a partitioned table. Fixes #5537

### CODE REVIEW

**Review by phd3 — COMMENTED** [review]
I think the properties mentioned in the WITH clause are connector specific. So not all connectors are required to support `partitioned_by`. For example, Iceberg connector has `partitioning` property instead. https://prestosql.io/docs/current/connector/iceberg.html 

For this reason, I feel the examples make sense in connector level doc, but not sure about top level create-table doc. Thoughts?

File: presto-docs/src/main/sphinx/sql/create-table.rst [hunk 1]
```diff
@@ -84,6 +84,15 @@ plus additional columns at the start and end::
       another_orderdate date
     )
 
+Create the table ``orders`` with partition::
```
Comment (mosabua) [inline presto-docs/src/main/sphinx/sql/create-table.rst:87]:
```
Create the table ``orders`` and partition the data by the content of the ``orderdate`` column.
```

Also need to explain why there is ARRAY['orderdate'] ..

**Comment by findepi** [comment]
> I think the properties mentioned in the WITH clause are connector specific.

correct

> So not all connectors are required to support `partitioned_by`.

"not all" is an understatement. Most does not.

The presentation problem is:

- how to make the overview page useful to end users (examples are awesome!)
- without fooling them into thinking that all connectors support same properties (examples can be misleading!)
- ... and telling them find more information when they need it

BTW this is similar problem to https://prestosql.io/docs/current/sql/analyze.html except there we support just one connector (today) and so we could cut  some corners.

cc @electrum

File: presto-docs/src/main/sphinx/sql/create-table.rst [hunk 2]
```diff
@@ -84,6 +84,18 @@ plus additional columns at the start and end::
...
 
+Create the table ``orders`` and partition the data by the content of
+the ``orderdate`` column::
+    CREATE TABLE orders (
+      orderkey bigint,
+      orderstatus varchar,
+      totalprice double,
+      orderdate date
+    )
+    WITH (partitioned_by = ARRAY['orderdate']);
```
Comment (tooptoop4) [inline presto-docs/src/main/sphinx/sql/create-table.rst:95]: partition by 2 columns as from this reader does not know whether it is ARRAY['orderdate','orderstatus'] or ARRAY['orderdate,orderstatus'] or ARRAY['orderdate'],ARRAY['orderstatus']

Comment (mosabua) [reply presto-docs/src/main/sphinx/sql/create-table.rst:95]: thats a good idea.. you then need to change the status column to be lower in the DDL as well.

File: presto-docs/src/main/sphinx/sql/create-table.rst [hunk 3]
```diff
@@ -84,6 +84,18 @@ plus additional columns at the start and end::
       another_orderdate date
     )
 
+Create the table ``orders`` and partition the data by the content of
+the ``orderdate`` column::
```
Comment (mosabua) [inline presto-docs/src/main/sphinx/sql/create-table.rst:88]: empty line after ::

File: presto-docs/src/main/sphinx/sql/create-table.rst [hunk 4]
```diff
@@ -84,6 +84,18 @@ plus additional columns at the start and end::
...
+    CREATE TABLE orders (
+      orderkey bigint,
+      orderstatus varchar,
+      totalprice double,
+      orderdate date
+    )
+    WITH (partitioned_by = ARRAY['orderdate']);
+Partition columns should be specified at last in the DDL. All the partition columns of table are specified in ARRAY.
```
Comment (mosabua) [inline presto-docs/src/main/sphinx/sql/create-table.rst:96]:
```
Partition columns need to be specified as the last columns in the DDL. All partition columns are specified in the ``ARRAY``.
```

File: presto-docs/src/main/sphinx/sql/create-table.rst [hunk 5]
```diff
@@ -84,6 +84,18 @@ plus additional columns at the start and end::
...
+The above table is partitioned on orderdate column.
```
Comment (mosabua) [inline presto-docs/src/main/sphinx/sql/create-table.rst:97]: Remove this here.. it is already stated above the sql snippet

**Review by mosabua — CHANGES_REQUESTED** [review]
just a bit of clean up needed

**Review by tooptoop4 — APPROVED** [review]

**Review by mosabua — APPROVED** [review]

**Review by phd3 — COMMENTED** [review]
I think the PR, in its current form, would be misleading, since the property is specific to hive connector. (In fact, the part `with (format = 'ORC')` in the first example of the current documentation has the same problem.) 

I can think of two options here:

1. Mention in "Description" section something like: "Please checkout the [connector documentation](https://prestosql.io/docs/347/connector.html) to look for examples on using connector specific table properties."

OR

2. Add the examples here by explicitly mentioning which connector the example corresponds to. e.g. something like `For hive connector, we can create table and partition the data ....`. 

I like the first option better, since it helps avoid connector-specific details on the top-level page that mainly attempts to provide information on what the syntax does. @findepi thoughts?

File: presto-docs/src/main/sphinx/sql/create-table.rst [hunk 6]
```diff
@@ -84,6 +84,19 @@ plus additional columns at the start and end::
...
+
+    CREATE TABLE orders (
+      orderkey bigint,
+      totalprice double,
+      orderdate date,
+      orderstatus varchar
+    )
+    WITH (partitioned_by = ARRAY['orderdate','orderstatus']);
+
+Partition columns need to be specified as the last columns in the DDL. All partition columns are specified in the ``ARRAY``.
```
Comment (findepi) [inline presto-docs/src/main/sphinx/sql/create-table.rst:98]: nit: do not use "DDL" abbreviation

**Review by findepi — COMMENTED** [review]
> I think the PR, in its current form, would be misleading

@phd3 i agree. 
It's useful to have an example here on this page, but it should explicitly say it pertains the Hive connector.

**Comment by mosabua** [comment]
:wave: @agrawalreetika  - this PR has become inactive. If you're still interested in working on it, please let us know, and we can try to get reviewers to help with that. Alternatively we can use your suggestion for a new PR.

We're working on closing out old and inactive PRs, so if you're too busy or this has too many merge conflicts to be worth picking back up, we'll be making another pass to close it out in a few weeks.

**Pull request closed without merging**